src/classes/UTIL_IntegrationConfig.cls

```diff
@@ -0,0 +1,147 @@
+    private static final String DEFAULT_NPSP_DSO_BRIDGE_NAMESPACE = 'dso_npsp';
+
+    @TestVisible
+    protected Boolean isInstalled {
+        get {
+            if (isInstalled == null) {
+                isInstalled = initIsInstalled();
+            }
+            return isInstalled;
+        }
+        set;
+    }
+    @TestVisible
+    protected String namespace {
+        get {
+            if (namespace == null) {
+                namespace = initNamespace();
+            }
+            return namespace;
+        }
+        set;
+    }
+
+    @TestVisible
+    protected String callableApiClassName;
+
+    @TestVisible
+    protected System.Callable callableApi {
+        get {
+            if (callableApi == null) {
+                callableApi = initCallableApi();
+            }
+            return callableApi;
+        }
+        set;
+    }
+
+    /**********************************************************************************************************
+    * @description Checks if the package is installed
+    */
+    protected abstract Boolean initIsInstalled();
+
+    /**********************************************************************************************************
+    * @description Determines the package namespace
+    */
+    protected abstract String initNamespace();
+
+    /**********************************************************************************************************
+    * @description Confirms if the (un)managed package is installed
+    */
+    public Boolean isInstalled() {
+        return isInstalled;
+    }
+
+    /**********************************************************************************************************
+    * @description Instantiate the Callable API endpoint in the package
+    */
+    public System.Callable getCallableApi() {
+        return callableApi;
+    }
+
+    /**********************************************************************************************************
+    * @description Instantiate the Callable API endpoint in the package
+    */
+    private System.Callable initCallableApi() {
+        if (!isInstalled) {
+            return null;
+        }
+
+        return (Callable) Type.forName(namespace, callableApiClassName).newInstance();
+    }
+
+    /**********************************************************************************************************
+    * @description Adds the namespace before an SObject name or a child relationship name
+    */
+    public String prefixNamespace(String entity) {
+        if (String.isBlank(namespace)) {
+            return entity;
+        }
+
+        return namespace + '__' + entity;
+    }
+
+    public class DsoNpspPackage extends UTIL_IntegrationConfig {
+        /**********************************************************************************************************
+        * @description Constructor
+        */
+        public DsoNpspPackage() {
+            callableApiClassName = 'BridgeCallable_API';
```

Comment:
What do you think about naming a class without underscores, ie `BridgeCallableApi` (or `BridgeCallableAPI`)?

Reply:
Actually, good catch, `BridgeCallableApi` is the name of the class in DSO-NPSP: https://github.com/SalesforceFoundation/DSO-NPSP/pull/4/files#diff-f65a45e6c2e8a4b24105d50c33c16292R13